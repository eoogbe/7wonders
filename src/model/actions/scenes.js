/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const VIEW_WONDER_CHOOSER = 'VIEW_WONDER_CHOOSER';

export const VIEW_WONDER_PANEL = 'VIEW_WONDER_PANEL';

export const VIEW_GAME_BOARD = 'VIEW_GAME_BOARD';

export const START_AGE = 'START_AGE';

export const SCORE_GAME = 'SCORE_GAME';

export const QUIT = 'QUIT';

export const viewWonderChooser = (playerId) => ({
  type: VIEW_WONDER_CHOOSER,
  payload: {
    playerId,
  },
});

export const viewWonderPanel = () => ({
  type: VIEW_WONDER_PANEL,
});

export const viewGameBoard = () => ({
  type: VIEW_GAME_BOARD,
});

export const startAge = () => ({
  type: START_AGE,
});

export const scoreGame = () => ({
  type: SCORE_GAME,
});

export const quit = () => ({
  type: QUIT,
});
