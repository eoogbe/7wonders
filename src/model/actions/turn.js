/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const BUILD_STRUCTURE = 'BUILD_STRUCTURE';

export const BUILD_WONDER = 'BUILD_WONDER';

export const ADD_MONEY = 'ADD_MONEY';

export const SELECT_HAND_CARD = 'SELECT_HAND_CARD';

export const CHOOSE_NEIGHBOR = 'CHOOSE_NEIGHBOR';

export const COPY_GUILD = 'COPY_GUILD';

export const buildStructure = (cardId) => ({
  type: BUILD_STRUCTURE,
  payload: {
    cardId,
  },
});

export const buildFreeStructure = () => ({
  type: BUILD_STRUCTURE,
  payload: {
    isFree: true,
  },
});

export const buildWonder = () => ({
  type: BUILD_WONDER,
});

export const addMoney = () => ({
  type: ADD_MONEY,
});

export const selectHandCard = (cardId, cardIdx) => ({
  type: SELECT_HAND_CARD,
  payload: {
    cardId,
    cardIdx,
  },
});

export const chooseNeighbor = (costType) => ({
  type: CHOOSE_NEIGHBOR,
  payload: {
    costType,
  },
});

export const copyGuild = (cardId) => ({
  type: COPY_GUILD,
  payload: {
    cardId,
  },
});
