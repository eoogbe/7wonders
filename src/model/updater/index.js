/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { concat, decorate, withDefaultState } from 'redux-fp';
import ages from './ages';
import app, { initialAppState } from './app';
import copyAiGuild from './copyAiGuild';
import hands from './hands';
import military from './military';
import neighbors from './neighbors';
import players from './players';
import takeAiTurns from './takeAiTurns';
import trade from './trade';
import wonder from './wonder';

const initialState = {
  ages: [],
  players: {},
  app: initialAppState,
};

export default decorate(
  withDefaultState(initialState),
  concat(
    wonder,
    neighbors,
    takeAiTurns,
    ages,
    hands,
    players,
    military,
    copyAiGuild,
    app,
    trade
  )
);
