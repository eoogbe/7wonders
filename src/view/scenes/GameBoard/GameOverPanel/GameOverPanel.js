/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { compose } from 'ramda';
import { useDispatch } from 'react-redux';
import * as actions from 'model/actions';
import QuitButton from 'view/components/QuitButton';
import * as theme from 'view/theme';
import './GameOverPanel.css';

const GameOverPanel = () => {
  const dispatch = useDispatch();
  const scoreGame = compose(dispatch, actions.scoreGame);

  const buttonStyle = {
    fontSize: '1.25em',
    width: '8em',
    backgroundColor: theme.primaryColor,
    color: '#FFF',
  };

  return (
    <div className="GameOverPanel">
      <QuitButton style={buttonStyle} />
      <button
        className="primary-button"
        style={buttonStyle}
        type="button"
        onClick={scoreGame}
      >
        View scores
      </button>
    </div>
  );
};

export default GameOverPanel;
