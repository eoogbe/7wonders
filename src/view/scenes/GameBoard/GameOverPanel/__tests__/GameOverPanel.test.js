/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { scoreGame } from 'model/actions';
import GameOverPanel from '../GameOverPanel';

const mockStore = configureStore();

it('renders without crashing', () => {
  const store = mockStore({});
  const tree =
    renderer.create(
      <Provider store={store}>
        <GameOverPanel />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('scores the game', () => {
  const store = mockStore({});
  const tree = mount(
    <Provider store={store}>
      <GameOverPanel />
    </Provider>
  );
  tree.find('button').last().simulate('click');
  expect(store.getActions()).toEqual([scoreGame()]);
});
