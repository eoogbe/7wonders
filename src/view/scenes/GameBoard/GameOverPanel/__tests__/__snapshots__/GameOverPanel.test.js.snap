// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="GameOverPanel"
>
  <button
    className="QuitButton"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(93, 64, 55)",
        "color": "#FFF",
        "fontSize": "1.25em",
        "width": "8em",
      }
    }
    type="button"
  >
    Quit
  </button>
  <button
    className="primary-button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(93, 64, 55)",
        "color": "#FFF",
        "fontSize": "1.25em",
        "width": "8em",
      }
    }
    type="button"
  >
    View scores
  </button>
</div>
`;
